weave/trace/weave_client.py

```diff
@@ -163,7 +169,7 @@ def map_to_refs(obj: Any) -> Any:
 class Call:
     """A Call represents a single operation that was executed as part of a trace."""
 
-    op_name: str
+    _op_name: Union[str, Future[str]]
```

Comment:
this doesn't feel right on a public api

Reply:
This is a good comment - i agree. Which is why it is underscored. The public API is just `.op_name`. Obviously this pattern is used throughout this PR. Do you have an alternative?

Reply:
You could use an alternate constructor, but I'm not sure it's much prettier here.

The alt would be something like:
```py
@classmethod
def from_future(cls, fut): ...
```